### ISSUE TITLE
Remove Logger usage into MP

### ISSUE THREAD

**Issue by Lomadriel** [Mirai-Team/mirai-project#70]
I think that we have to remove it usage into our Library.

We could use [cerr](http://www.cplusplus.com/reference/iostream/cerr/?kw=cerr) or [sf::err](http://www.sfml-dev.org/documentation/2.3/group__system.php#ga7fe7f475639e26334606b5142c29551f). By using this, you can redirect errors on a file or simply display them in stdout stream.

**Comment by Lomadriel** [comment]
I wait a decision before starting to enhance the Logger. (see #69)

**Comment by CBenoit** [comment]
I agree. But it's quite similar to what is already done (the main difference is that, currently, streams are not redirect to std::cerr, which could be a Logger class improvement).

Moreover, and it should have its own issue, some parts can use `asserts` and `throw` to handle exceptions.

**Comment by Lomadriel** [comment]
Yes, I agree but we should allow users to choose where they want to write or print MP errors or warnings.

[CBenoit](https://github.com/Mirai-Team/mirai-project/issues/70#issuecomment-111563371) said:

> (the main difference is that, currently, streams are not redirect to std::cerr, which could be a Logger class improvement).

The new Logger will indeed allow this.

**Comment by CBenoit** [comment]
Yes. You can add that here : https://github.com/Mirai-Team/mirai-project/issues/69

**Comment by Lomadriel** [comment]
With the new logger we can set a default stream in a static variable to log warnings or errors.

This variable is equal by default to std::cout.rdbuf() but users can replace it by what they want.
